contrib_bots/lib/giphy.py

```diff
@@ -0,0 +1,52 @@
+import requests
+
+
+class GiphyHandler(object):
+    '''
+    This plugin facilitates giving a gif image that appropriate with the
+    keywords user input. The gif image and search algorithm are provided by
+    giphy. It looks for messages starting with "@giphy", then we send an gif
+    image that appropriate with the keywords.
+    '''
+
+    GIPHY_API_KEY = 'dc6zaTOxFJmzC'  # This is public giphy API key.
+
+    def usage(self):
+        return '''
+            This plugin will allow users to insert gif images from giphy that appropriate
+            with the keywords user input. Users should preface keywords with "@giphy".
+            '''
+
+    def get_url_gif_giphy(self, keyword, api_key):
+        # Return gif giphy url based on keywords given.
+        query = {'s': keyword,
+                 'api_key': api_key}
+
+        data = requests.get('http://api.giphy.com/v1/gifs/translate', params=query)
+        return data.json()['data']['images']['downsized']['url']
+
+    def triage_message(self, message):
+        # Return True if we want to (possibly) response to this message.
+        original_content = message['content']
+        is_giphy_called = original_content.startswith('@giphy ')
+
+        return is_giphy_called
+
+    def handle_message(self, message, client, state_handler):
+        keyword = message['content'][7:]  # Selects chars that have index more than 7 (to strip).
+        try:
+            gif_url = self.get_url_gif_giphy(keyword, self.GIPHY_API_KEY)
+            new_content = '[Click to enlarge](%s)' % (gif_url)
```

Comment:
This looks good, but Giphy requires an attribution for production deployments. How would you address that?
https://github.com/Giphy/GiphyAPI#request-a-production-key